sign.js

```diff
@@ -327,29 +333,43 @@ var signAsync = module.exports.signAsync = function (opts) {
         if (!opts['entitlements-inherit']) {
           filePath = path.join(__dirname, 'default.entitlements.mas.inherit.plist')
           debugwarn('No `entitlements-inherit` passed in arguments:', '\n',
-            '* Sandbox entitlements file for enclosing app files is default to:', filePath)
+            '* Sandbox entitlements file for enclosed app files is default to:', filePath)
           opts['entitlements-inherit'] = filePath
         }
+        if (!opts['entitlements-loginhelper']) {
+          // Default to App Sandbox enabled
+          filePath = path.join(__dirname, 'default.entitlements.mas.plist')
+          debugwarn('No `entitlements-loginhelper` passed in arguments:', '\n',
+            '* Sandbox entitlements file for login helper is default to:', filePath)
+          opts['entitlements-loginhelper'] = filePath
+        }
       } else {
         // Not necessary to have entitlements for non Mac App Store distribution
         if (!opts.entitlements) {
           debugwarn('No `entitlements` passed in arguments:', '\n',
             '* Provide `entitlements` to specify entitlements file for codesign.')
         } else {
-          // If entitlements is provided as a flag, fallback to default
+          // If entitlements is provided as a boolean flag, fallback to default
           if (opts.entitlements === true) {
             filePath = path.join(__dirname, 'default.entitlements.darwin.plist')
             debugwarn('`entitlements` not specified in arguments:', '\n',
               '* Provide `entitlements` to specify entitlements file for codesign.', '\n',
-              '* Sandbox entitlements file for enclosing app files is default to:', filePath)
+              '* Entitlements file is default to:', filePath)
             opts.entitlements = filePath
           }
           if (!opts['entitlements-inherit']) {
             filePath = path.join(__dirname, 'default.entitlements.darwin.inherit.plist')
             debugwarn('No `entitlements-inherit` passed in arguments:', '\n',
-              '* Sandbox entitlements file for enclosing app files is default to:', filePath)
+              '* Entitlements file for enclosed app files is default to:', filePath)
             opts['entitlements-inherit'] = filePath
           }
+          if (!opts['entitlements-loginhelper']) {
+            // Default to App Sandbox enabled
+            filePath = path.join(__dirname, 'default.entitlements.mas.plist')
```

Comment:
Should this be the empty `default.entitlements.darwin.plist`? Probably shouldn't sign with sandbox by default if this is not a mas build.

Reply:
That's a good point 🤔

Since the apps that are not going to be shipped on the Mac App Store can still be sandboxed, we may need to take into consideration too if app sandbox is enabled for the Electron app and then selectively apply sandbox for the login helper by default too?

If users aren't enabling app sandbox on their Electron apps for distribution outside the Mac App Store we can fallback to empty entitlements?

Reply:
Isn't that then up to the user to provide a custom `entitlements-loginhelper`? If they are sandboxing their app then they must have already passed `entitlements` or `entitlements-inherit` and be familiar with the custom plist API.

Reply:
That sounds like a good idea 👌 
So it'll either be the same entitlements to sign the app bundle or a specifically specified entitlements for the login helper.